### PR TITLE
Issue 275 create settings

### LINKED ISSUE #282: Complete the population characteristics bug fixes
As explained by @ciremusyoka  [here](https://docs.google.com/document/d/1XdaL6pft6OY9wa8H8_1gEr0Unxw0V50Zu6y61tQqdyk/edit?usp=sharing), ANC needs the following things fixed for the pop-characteristics work

- [ ] If a POST is performed remove the `inherited from` label and put an `_`.

- [ ] If `Inherited from` id doesn’t give back an object show the `UUID` of the country it is inherited from

- [ ] If the inherit setting is clicked then a `delete` action should be done. the endpoint to hit is [https://anc-stage.smartregister.org/opensrp/rest/v2/settings/{id}](https://anc-stage.smartregister.org/opensrp/rest/v2/settings/{id})   where the `id`  is the `settingMetadataId`  e.g `"settingMetadataId": "1277"`

- [ ] Update tests

LOE: 2 hours

### PR DESCRIPTION
**Pull request by kelvin-muchiri**
Closes #282 

Complete the population characteristics bug fixes

The following are the changes after the fixes

### Top Level location does not show **Inherit Setting**

Screenshot from 2020-08-06 19-51-37

### Lower levels show **Inherit Setting**. Clicking on **Inherit Setting** makes an HTTP DELETE API call

Screenshot from 2020-08-06 19-51-54

### Children locations show UUID of the parent if inherited from is missing

Screenshot from 2020-08-06 19-52-12

### CODE REVIEW

File: packages/populationCharacteristics/src/components/EditSettings/index.tsx [hunk 1]
```diff
@@ -165,21 +182,51 @@ const EditSetings = (props: FormConfigProps & EditSettingsDefaultProps) => {
...
-            .update(data)
-            .then(() => {
-                fetchSettings([{ ...row, value }], activeLoc);
-            })
-            .catch(error => {
-                customAlert && customAlert(String(error), { type: 'error' });
-            })
-            .finally(() => setLoading(false));
+
+        if (value === 'inherit') {
```
Comment (moshthepitt) [inline packages/populationCharacteristics/src/components/EditSettings/index.tsx:190]: Can we move this string to a named constant?

File: packages/populationCharacteristics/src/components/EditSettings/tests/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -1,15 +1,384 @@
 // Jest Snapshot v1, https://goo.gl/fbAQLP
```
Comment (moshthepitt) [inline packages/populationCharacteristics/src/components/EditSettings/tests/__snapshots__/index.test.tsx.snap:1]: This snapshot is too large.

File: packages/populationCharacteristics/src/ducks/settings/index.ts [hunk 3]
```diff
@@ -74,7 +74,9 @@ export default function reducer(state = initialState, action: LocSettingsTypes):
                 ...state,
                 settingsByLocId: {
                     ...state.settingsByLocId,
-                    [locId]: { ...(state.settingsByLocId as any)[locId], ...action.settingsByLocId[locId] },
+                    [locId]: action.replace
```
Comment (moshthepitt) [inline packages/populationCharacteristics/src/ducks/settings/index.ts:77]: We need to add a  test for this.

**Comment by ciremusyoka** [comment]
< Lower levels show Inherit Setting. Clicking on Inherit Setting makes an HTTP DELETE API call

@kelvin-muchiri do we have a confirmation dialog for the delete action?

**Comment by kelvin-muchiri** [comment]
> do we have a confirmation dialog for the delete action?

@ciremusyoka I don't think the user should be aware of the delete action. The feedback given to the user for this action is that the **Inherited from** value should now the label of the parent, and the value should change

File: packages/populationCharacteristics/src/components/EditSettings/index.tsx [hunk 4]
```diff
@@ -258,30 +263,55 @@ const EditSetings = (props: FormConfigProps & EditSettingsDefaultProps) => {
...
+
+                if (label) {
+                    inheritedFrom = label;
+                } else {
+                    inheritedFromInvalid = true;
+                }
+            } else {
+                inheritedFrom = '_';
+            }
+
```
Comment (ciremusyoka) [inline packages/populationCharacteristics/src/components/EditSettings/index.tsx:291]: Is it possible to have this extracted as a function?

File: packages/populationCharacteristics/src/ducks/locations/tests/utils.test.ts [hunk 5]
```diff
@@ -0,0 +1,193 @@
...
+                                tags: ['School'],
+                                voided: false,
+                            },
+                            parent: '44de66fb-e6c6-4bae-92bb-386dfe626eba',
+                        },
+                    },
+                    parent: '4e188e6d-2ffb-4b25-85f9-b9fbf5010d40',
+                },
+            },
+            parent: 'b1ef8a0b-275b-43fc-a580-1e21ceb34c78',
```
Comment (ciremusyoka) [inline packages/populationCharacteristics/src/ducks/locations/tests/utils.test.ts:165]: lets have his a variable in the fixtures

File: clients/core/package.json [hunk 6]
```diff
@@ -20,7 +20,7 @@
         "@onaio/superset-connector": "^0.0.13",
         "@opensrp/form-config": "^0.0.6",
         "@opensrp/opensrp-table": "^0.0.2",
-        "@opensrp/population-characteristics": "^0.0.6",
+        "@opensrp/population-characteristics": "file:../../packages/populationCharacteristics",
```
Comment (ciremusyoka) [inline clients/core/package.json:23]: Point the package to the previous version `^0.0.6` and update once the package is published.

File: packages/populationCharacteristics/src/ducks/settings/index.ts (hunk 3 above)
Comment (kelvin-muchiri) [reply packages/populationCharacteristics/src/ducks/settings/index.ts:77]: @moshthepitt These tests were added

**Review by moshthepitt — APPROVED** [review]

**Pull request merged**